City-of-Helsinki/drupal-hdbt
# UHF-8242: Workaround map limits

# [UHF-8242](https://helsinkisolutionoffice.atlassian.net/browse/UHF-8242)
Prevent map from breaking when search yields too much results.

## What was done
* Move map to its own component
* Query palvelukartta directly if result count is too high

## How to install

* Set up KASKO https://github.com/City-of-Helsinki/drupal-helfi-kasvatus-koulutus
* Install this branch: `composer install drupal/hdbt:UHF-8242-workaround-map-limits`
* Add this to your .evn file: `ELASTIC_PROXY_URL=https://kasvatus-elastic-proxy-test.agw.arodevtest.hel.fi` (testing with data from test server)
* `make up`
* `drush cr`

## How to test

* Edit or create a landing page to include a `school search` paragraph
* View the page. Click on `Show results as a map` -button on the right top corner of the results
* The map should load all the schools. This might take a few seconds.



[UHF-8242]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-8242?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
